### DISCUSSION TITLE
Adding touch support for parallel displays?

### DISCUSSION THREAD

**Discussion by JimDrewGH** [Bodmer/TFT_eSPI#1963] [Q&A - Touch] [answered]
Since a LOT of the new displays coming out of China are parallel with XPT2046 touch controllers, what are the chances of having the ability to have TFT_eSPI support touch with a parallel display?  Right now, compiling states that parallel+touch is not supported.

**Comment by Bodmer** [comment, marked as answer]
I am rather busy with paid work so this will have to wait some time.

Can you provicde some links to the parallel display vendors.

You can use third party touch libraries but any touch related sketches will need to be adapted.

**Reply by JimDrewGH** [reply]
Of course there are dozens on Aliexpress, but www.buydisplay.com is a good source.  They have several dozen different size displays with the XPT2046 touch + parallel.  I have several of their displays here including this one:

https://www.buydisplay.com/5-inch-tft-ssd1963-lcd-module-optl-touch-screen-display-480x272-arduino

They make a 800x400 display in the same exact form factor as well.  This is why I wanted parallel DMA.  :)

Really nice displays.

**Reply by Bodmer** [reply]
I have one of those displays with the capacitive touch screen, which I recommend as it is much more sensitive to touch and support multi touch. I wrote a support library here for that touch controller:
https://github.com/Bodmer/Touch_FT5x06/tree/main/examples/Basic_touch

Third part libraries like this will be needed until I can find the time to extend the capability of TFT_eSPI.

**Reply by Bodmer** [reply]
Just remembered this library which supports XPT2046 touch controllers and works with TFT_eSPI, it has an example:
https://github.com/achillhasler/TFT_eTouch

It has very good false touch and spurious touch filtering.

**Reply by JimDrewGH** [reply]
Perfect!  I will check that library out.  We can't use capacitive touch screens in most industrial environments because gloves are worn.  Resistive touch displays are the only thing that will work.
